prow/cmd/deck/job_history.go

```
@@ -0,0 +1,291 @@
+}
+
+type finished struct {
+	Timestamp int64  `json:"timestamp"`
+	Result    string `json:"result"`
+}
+
+func readLatestBuild(obj *storage.ObjectHandle) (int, error) {
+	rc, err := obj.NewReader(context.Background())
+	if err != nil {
+		return -1, fmt.Errorf("Failed to get reader for latest build number: %v", err)
+	}
+	data, err := ioutil.ReadAll(rc)
+	if err != nil {
+		return -1, fmt.Errorf("Failed to read latest build number: %v", err)
+	}
+	n64, err := strconv.Atoi(string(data))
+	if err != nil {
+		return -1, fmt.Errorf("Failed to parse latest build number: %v", err)
+	}
+	return int(n64), nil
+}
+
+// resolve sym links into the actual log directory for a particular test run
+func resolveSymLink(bkt *storage.BucketHandle, symLink string) (string, error) {
+	linkObj := bkt.Object(symLink)
+	rc, err := linkObj.NewReader(context.Background())
+	if err != nil {
+		return "", err
+	}
+	data, err := ioutil.ReadAll(rc)
+	if err != nil {
+		return "", err
+	}
+	bAttrs, err := bkt.Attrs(context.Background())
+	if err != nil {
+		return "", err
+	}
+	// strip gs://<bucket-name> from global address `u`
+	u := string(data)
+	i := strings.Index(u, bAttrs.Name)
+	dir := u[i+len(bAttrs.Name)+1:] // +1 for leading slash
+	return dir, nil
+}
+
+func getDir(bkt *storage.BucketHandle, root, id string) string {
+	bAttrs, err := bkt.Attrs(context.Background())
+	if err != nil {
+		return ""
+	}
+	bktName := bAttrs.Name
+	if strings.HasPrefix(root, logsPrefix) {
+		return path.Join(bktName, root, id)
+	}
+	symLink := path.Join(root, id+".txt")
+	dir, err := resolveSymLink(bkt, symLink)
+	if err != nil {
+		return ""
+	}
+	return path.Join(bktName, dir)
+}
+
+func getObject(bkt *storage.BucketHandle, root, id, fname string) *storage.ObjectHandle {
+	if strings.HasPrefix(root, logsPrefix) {
+		p := path.Join(root, id, fname)
+		return bkt.Object(p)
+	}
+	symLink := path.Join(root, id+".txt")
+	dir, err := resolveSymLink(bkt, symLink)
+	if err != nil {
```

Don't swallow the error. Either return it, or handle and log the it, whichever is more appropriate.

---

this looks to be fixed.